versionutils/version.go

```diff
@@ -15,6 +15,7 @@ import (
 const (
 	SemverNilVersionValue = "v0.0.0"
 	SemverMinimumVersion  = "v0.0.1"
+	wasmLabel             = "wasm"
```

Comment:
can we make this more generic? i'd like to be able to support tags that have generic secondary labels, e.g.:
- 1.0.0
- 1.0.0-rc1
- 1.0.0-wasm-rc1
- 1.0.0-beta1-foobar

Reply:
i think we should just make the label version everything that comes after the initial hyphen, as this supports arbitrary amounts of hyphens following a version and doesn't require changes to our version comparison logic

Reply:
spoke offline, allowing labels with any amount of hyphens and stripping out any numbers to get the label version number